### ISSUE TITLE
Is it possible to reproduce only one HEVC tile?

### ISSUE THREAD

**Issue by henriquedgarcia** [gpac/gpac#1213]
Thanks for reporting your issue. Please make sure these boxes are checked before submitting your issue - thank you!

- [ X] I looked for a similar issue and couldn't find any.
- [ X] I tried with the latest version of GPAC. Installers available at http://gpac.io/downloads/gpac-nightly-builds/
- [ X] I give enough information for contributors to reproduce my issue (meaningful title, github labels, platform and compiler, command-line ...). I can share files anonymously with this dropbox: https://www.mediafire.com/filedrop/filedrop_hosted.php?drop=eec9e058a9486fe4e99c33021481d9e1826ca9dbc242a6cfaab0fe95da5e5d95

Detailed guidelines: http://gpac.io/2013/07/16/how-to-file-a-bug-properly/
---------------------------------------------------------------------------------------

Hi. I'm doing a college job and would like to play only one tile from a 360° HEVC tiled video, but MP4Client show several errors.

After following the instructions at https://gpac.wp.imt.fr/2017/02/01/hevc-tile-based-adaptation-guide/ to produce a tiled HEVC video I would like to play only one tile.

I codify the video using
`kvazaar --input video.yuv --input-res 4320x2160 --input-fps 30 -p 30 --tiles 3x3 --slices tiles --mv-constraint frametilemargin --bitrate 3000000 --output video.hevc`

so I add in a MP4 file using MP4Box:
`MP4Box -add video.hevc: split_tiles -new video.mp4`

The video.mp4 file plays perfectly using
`MP4Client video.mp4`

However, when I extract a track using
`MP4Box -raw TrackID video_tiled.mp4`

All .hvc files fails. Only track 2 appears an image, but it does not play. All other tracks fail and display the message:
> $ MP4Client video_track3.hvc
> Using config file in /home/henrique/.gpac directory
> System info: 15912 MB RAM - 8 colors
> Modules Found: 41
> Loading GPAC Terminal
> [Thread MediaManager] Could not set priority (2) for thread ID 0x9effd700
> [Thread MediaManager] Could not set priority (2) for thread ID 0x9effd700
> Terminal Loaded in 139 ms
> Opening URL video_track3.hvc
> [hevc @ 0x5613e8d85b40] missing picture in access unit
> [hevc @ 0x5613e8d85b40] PPS changed between slices.
> [hevc @ 0x5613e8d85b40] Error parsing NAL unit # 3.
> [hevc @ 0x5613e8d84b40] decoding for stream 0 failed
> [hevc @ 0x5613e8d84b40] Could not find codec parameters for stream 0 (Video: hevc (Main), none): unspecified size
> Consider increasing the value for the 'analyzeduration' and 'probesize' options
> [NULL @ 0x5613e8d85b40] missing picture in access unit
> [hevc @ 0x5613e8d97ae0] missing picture in access unit
> [hevc @ 0x5613e8d97ae0] PPS changed between slices.
> [hevc @ 0x5613e8d97ae0] Error parsing NAL unit # 3.
> [hevc @ 0x5613e8d85680] decoding for stream 0 failed
> [hevc @ 0x5613e8d85680] Could not find codec parameters for stream 0 (Video: hevc (Main), none): unspecified size
> Consider increasing the value for the 'analyzeduration' and 'probesize' options
> Service Connected
> [hevc @ 0x7fa680030060] PPS changed between slices.
> [hevc @ 0x7fa680030060] Error parsing NAL unit # 3.
> Service Disconnected
> Deleting terminal ... done (in 123 ms) - ran for 5708 ms
> GPAC cleanup ...

So, I removed the unwanted tiles using the _-rem_ option. For example to leave only track 2 (tile 1) I use
`MP4Box -rem 3 -rem 4 -rem 5 -rem 6 -rem 7 -rem 8 -rem 9 -rem 10 video.mp4 -out video_2.mp4`

But this works correctly for track 2 only. When I use this for other tracks like this:
`MP4Box -rem 2 -rem 4 -rem 5 -rem 6 -rem 7 -rem 8 -rem 9 -rem 10 video.mp4 -out video_3.mp4`

The MP4Client fails and displays the error:

> $ MP4Client video_3.mp4
> Using config file in /home/henrique/.gpac directory
> System info: 15912 MB RAM - 8 colors
> Modules Found: 41
> Loading GPAC Terminal
> [Thread MediaManager] Could not set priority (2) for thread ID 0xb97fa700
> [Thread MediaManager] Could not set priority (2) for thread ID 0xb97fa700
> Terminal Loaded in 129 ms
> Opening URL video_3.mp4
> Service Connected
> [HEVC @ ffmpeg] Initializing with 7 threads
> [hevc @ 0x7f8c8c0871e0] PPS changed between slices.
> [hevc @ 0x7f8c8c0871e0] Error parsing NAL unit # 5.
> [hevc @ 0x7f8c8c094880] PPS changed between slices.
> [hevc @ 0x7f8c8c094880] Error parsing NAL unit # 0.
> [hevc @ 0x7f8c8c0b3680] PPS changed between slices.
> [hevc @ 0x7f8c8c0b3680] Error parsing NAL unit # 0.
> ...
> Service Disconnected
> Deleting terminal ... done (in 111 ms) - ran for 5448 ms
> GPAC cleanup ...

Is it really possible to play only one tile?
I'm using Ubuntu 18.04.2 and GPAC version 0.7.2-DEV-rev1018-g0ab801f8b-master and Kvazaar v1.2.0-244-ga13fc510 2019-02-04. 

Later, I also installed an external OpenHEVC in Gepac build using this instructions (https://github.com/gpac/gpac/issues/979) and there was no result.

Thanks.

**Comment by henriquedgarcia** [comment]
I recently installed GPAC for Windows and performed the same test above. By keeping in the MP4 container only the track of interest using the command:
`mp4box.exe -rem 3 -rem 6 -rem 4 -rem 5 -rem 7 -rem 8 -rem 9 -rem 2. \ video.mp4 -out. \ video_only10.mp4`

So I tried to play the video using the command:
`mp4client.exe. \ video_only10.mp4`

To my surprise the video was reproduced correctly. So I tried to play on Windows some MP4 files that were manipulated in Linux and also played successful. But when I try to play using MP4Client under linux, playback fails. Apparently this problem only affects Linux MP4Client.

However, when I try to remove a tab using the
mp4box.exe -raw 10 video.mp4

The resulting file (video_track10.hvc) continues to display the same error when using mp4client under Windows.
